Add VSCode Postgres Extension, splice in migration docs (#12192)
PrairieLearn/PrairieLearn: docs/dev-guide/guide.md

## Database stored procedures (sprocs)

!!! warning

>     We are migrating away from the use of [sprocs](https://www.geeksforgeeks.org/postgresql/postgresql-introduction-to-stored-procedures/). Prefer writing logic in TypeScript instead.

Reviewer: I have somewhat of a reflexive aversion to geeksforgeeks. Can we link to something else, either the Postgres docs directly or something like https://neon.com/postgresql/postgresql-plpgsql/postgresql-create-procedure if you want a higher-level description?

Author: Done in https://neon.com/postgresql/postgresql-plpgsql/postgresql-create-procedure

- For each variant of a question that a student sees they will have submitted zero or more `submissions` with a `variant_id` to show what it belongs to. The submissions row also contains information the submitted answer and whether it was correct.

??? tip "Schema visualization"

    Using the `ms-ossdata.vscode-pgsql` extension, you can visualize the schemas within VSCode.

    ![Setup Postgres VSCode](./postgres-vscode-setup.png)
    ![Using Postgres VSCode](./postgres-vscode-tool.png)

??? tip "Removing old schemas"

>     The following query will remove all schemas except public and the most recent one.

Reviewer: This is clever, but is probably unnecessarily complex. Maybe we should instead document a script that removes everything but `public` and advise folks to just stop their server before cleaning and start it again after.

Author: Complex as in we might have to change it in the future? If you think that is the case, I'll update the recommendation.

Reviewer: Complex as in inherently complex for not much value: there is no point in retaining even the latest schema in anything besides production environments, as restarting the server will create a new one automatically.

- For each variant of a question that a student sees they will have submitted zero or more `submissions` with a `variant_id` to show what it belongs to. The submissions row also contains information the submitted answer and whether it was correct.

> ??? tip "Schema visualization"

Reviewer: ```suggestion
??? tip "Schema and data exploration"
```

>     Using the `ms-ossdata.vscode-pgsql` extension, you can visualize the schemas within VSCode.

Reviewer: ```suggestion
    The [`ms-ossdata.vscode-pgsql` VSCode extension](https://marketplace.visualstudio.com/items?itemName=ms-ossdata.vscode-pgsql) can help you explore the database schema and data in your editor.
```

    ![Setup Postgres VSCode](./postgres-vscode-setup.png)

    ![Using Postgres VSCode](./postgres-vscode-select-1000.png)

??? tip "Cleaning out old schemas"

>     The following query will remove all schemas except public. You can then restart the server to recreate the sprocs.

Reviewer: ```suggestion
    The following query will remove all schemas except `public`. You can then restart the server to recreate the sprocs.
```

## Database schema modification

> See the [migrations page](./migrations/README.md).

Reviewer: I'm really really not sure about this. In isolation, there's no way to know what file this points to without understanding how the docs are build.

_Do_ we just have to know that, and rely on always reading the dev guide as published in the docs?

Author: I always read the dev guide as published in the docs, and I found the click away a little annoying, especially from a discoverability perspective.

Author: I would be curious if anyone reads the dev guide outside of the docs.

Reviewer: To be fair, I only really do when I'm editing it. I guess this is optimizing for consumption rather than authoring?

```suggestion
See the [migrations documentation](./migrations/README.md).
```

Author: Yep!
